go/runtime/client/watcher.go

```diff
@@ -127,8 +129,19 @@ func (w *blockWatcher) watch() {
 			w.watched[*newWatch.id] = newWatch
 			if conn := w.committeeClient.GetConnection(); conn != nil {
 				client := txnscheduler.NewTransactionSchedulerClient(conn)
+				var epochNumber api.EpochTime
+				epochNumber, err = w.common.consensus.EpochTime().GetEpoch(w.common.ctx, height)
```

Comment:
In this case `height` will always be zero, so this will actually be incorrect.

You should create a function-local variable called `currentEpoch` (outside the for loop) and update it only below after processing an epoch transition block. Here, you just pass `currentEpoch` instead of doing any other queries.